### PR TITLE
feat: add useAxios hook  and tests

### PR DESCRIPTION
**Pull request by archit7448**
### Issue 
- Write useAxios hook and  tests for it.

### Files  to change
- ```jest.config.js```  for test setup
-  ```package.json```
      - Update react and react-testing-library version 
      - Add msw and axios
      - Inside scripts add test-coverage command.
- ```src/custom-hooks/useFetch.js```  change URL methods of passing change with props.

### Files to add 
- ```setupTests.ts```  add teardown
- ```mocks/server.ts``` add server
- ```mocks/handler.ts``` add handler
-  ```hooks/useAxios.ts``` feat: useFetch hook.
- ```__tests__/useAxios.ts``` test for useFetch hook.
  
 ### UseAxios hook 

 - The `useAxios` hook is a custom React hook for making HTTP requests with the `axios` library. It returns an object with three properties: `data`, `loading`, and `error`. The `data` property contains the response data, the `loading` property is a boolean that indicates whether the request is in progress, and the `error` property contains any error that occurred during the request.
 - The hook accepts a URL, an HTTP method (defaults to `GET`), and a configuration object that can be used to pass additional options to `axios`
 
### UseAxios hook test 

- The tests for the `useAxios` hook use the `msw` library to mock HTTP requests and verify that the hook returns the expected values for different scenarios, such as successful requests and failed requests.
- `get` , `patch` and `reject` request during testing.

### CODE REVIEW

File: src/hooks/useAxios.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const intialState: UseAxiosResult = {
+    data: null,
+    loading: true,
+    error: null,
+  };
+
+  const [state, setState] = useState(intialState);
+
+  useEffect(() => {
+    const request = axios[method](url, {
```
Comment (ivinayakg) [inline src/hooks/useAxios.ts:26]: how will the post, put, patch request work?
Will the data be sent in the config only or what?

File: my-app/package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "my-app",
+  "version": "0.1.0",
+  "private": true
+}
```
Comment (ivinayakg) [inline my-app/package.json:5]: what is this?
why do we have to create this file?

**Review by ivinayakg — CHANGES_REQUESTED** [review]
you have both package-lock.json as well as yarn.lock files in your PR which should not be the case.
Delete the package-lock file, apart from that there are a few other comments that I added.

**Comment by ivinayakg** [comment]
First, get #7 PR merge and then add this PR, both have the same code just one file is changed.
Get the first PR merged first and then raise a PR for `ueAxios` feature 🙂

File: my-app/package.json (hunk 2 above)
Comment (archit7448) [reply my-app/package.json]: By mistake it happens I have checked the app it work fine without it.

File: src/hooks/useAxios.ts (hunk 1 above)
Comment (archit7448) [reply src/hooks/useAxios.ts:26]: This way 
```useAxios(patchUrl, "patch", {
        data: {
          patch: ["members-site", "real-dev", "real-dev-squad", "patch"],
        },
        withCredentials: true,
      })
    );```

**Pull request closed without merging**